core/src/torii/utils.rs

```diff
@@ -25,13 +25,13 @@ impl<T: Encode + Send> Reply for Scale<T> {
 }
 
 /// Adds state to filter
-pub fn add_state<State: Send + Clone>(
-    state: State,
-) -> impl Filter<Extract = (State,), Error = Rejection> + Clone + Send {
-    warp::any().and_then(move || {
-        let state = state.clone();
-        async move { Ok::<_, Rejection>(state) }
-    })
+macro_rules! add_state {
```

Comment:
Any particular reason to use a declarative macro instead of a templated function?

Reply:
I don't think it's possible to create variadic functions in rust

Reply:
https://github.com/rust-lang/rust/issues/44930

yeah. It's not been finished yet (much less stabilised).

Reply:
Yeah, macros are pretty much the standard way to handle these cases in Rust at the moment.